src/cls/rbd/cls_rbd_types.h

```diff
@@ -216,6 +217,104 @@ struct GroupSpec {
 
 WRITE_CLASS_ENCODER(GroupSpec);
 
+enum SnapshotNamespaceType {
+  SNAPSHOT_NAMESPACE_TYPE_USER = 0,
+  SNAPSHOT_NAMESPACE_TYPE_GROUP = 1
+};
+
+struct UserSnapshotNamespace {
+  static const uint32_t SNAPSHOT_NAMESPACE_TYPE = SNAPSHOT_NAMESPACE_TYPE_USER;
+
+  UserSnapshotNamespace() {}
+
+  void encode(bufferlist& bl) const {}
+  void decode(bufferlist::iterator& it) {}
+
+  void dump(Formatter *f) const {}
+
+  inline bool operator==(const UserSnapshotNamespace& usn) const {
+    return true;
+  }
+
+};
+
+WRITE_CLASS_ENCODER(UserSnapshotNamespace);
```

Comment:
Minor: since these will always be encoded via ```SnapshotNamespaceEntry```, no need to create encode/decode free functions for these types.